lib/hanami/configuration.rb

```diff
@@ -0,0 +1,205 @@
+# frozen_string_literal: true
+
+require "uri"
+require "concurrent/hash"
+require "concurrent/array"
+require "dry/inflector"
+
+module Hanami
+  # Hanami application configuration
+  #
+  # @since 2.0.0
+  #
+  # rubocop:disable Metrics/ClassLength
+  class Configuration
+    require "hanami/configuration/cookies"
+    require "hanami/configuration/sessions"
+    require "hanami/configuration/middleware"
+    require "hanami/configuration/security"
+
+    # rubocop:disable Metrics/MethodLength
+    def initialize(env:)
+      @settings = Concurrent::Hash.new
+
+      self.env = env
+      self.environments = DEFAULT_ENVIRONMENTS.dup
+
+      self.base_url = DEFAULT_BASE_URL
+
+      self.logger   = DEFAULT_LOGGER.dup
+      self.routes   = DEFAULT_ROUTES
+      self.cookies  = DEFAULT_COOKIES
+      self.sessions = DEFAULT_SESSIONS
+
+      self.default_request_format  = DEFAULT_REQUEST_FORMAT
+      self.default_response_format = DEFAULT_RESPONSE_FORMAT
+
+      self.middleware = Middleware.new
+      self.security   = Security.new
+
+      self.inflections = Dry::Inflector.new
+    end
+    # rubocop:enable Metrics/MethodLength
+
+    def finalize
```

Comment:
Because `Hanami::Container` is subclassing `Dry::System::Container`, which exposes `.finalize` as a booting trigger, for consistency sake, I named this method the same.

Given this is a private method, I don't think people will look at it, nor confuse with Ruby's object finalization.